### PR TITLE
Cleanup for `apt::source`

### PR DESCRIPTION
**Pull request by mhaskel**
* Update `release` to default to `$::lsbdistcodename`
* Default `include_src` to false
* Validate more things!
* Stop redefining variables from `apt::params`

### CODE REVIEW

**Comment by mhaskel** [comment]
More will need to be done on `apt::source` once #428 is merged

**Comment by mhaskel** [comment]
Also, I don't know if we still want to turn key_* into a hash since we can't use the splats until 3.x.

File: manifests/source.pp [hunk 1]
```diff
@@ -17,25 +16,16 @@
...
-  } else {
-    $release_real = $release
+  if ! $release {
+    fail('lsbdistcodename fact not available: release parameter required')
   }
 
   file { "${name}.list":
     ensure  => $ensure,
-    path    => "${sources_list_d}/${name}.list",
+    path    => "${apt::params::sources_list_d}/${name}.list",
```
Comment (daenney) [inline manifests/source.pp]: Can't we use `$::apt::sources_list_d`?

**Comment by mhaskel** [comment]
This includes commit from #435.

**Pull request merged**